src/main/java/core/basesyntax/BooleanExpression.java

```diff
@@ -14,6 +14,9 @@
      * без использования конструкций if-else</p>
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+
+        return (a == b && a == c && a == d) ? false
+                : ((a ^ b) ^ (c ^ d)) ? false
+                : true;
```

Comment:
good job, but you can make it even shorter, try to avoid ternary operator

Reply:
done